src/main/java/org/apache/commons/lang3/RegExUtils.java

```diff
@@ -751,4 +755,67 @@ private static String toStringOrNull(final CharSequence text) {
     public RegExUtils() {
         // empty
     }
+
+    /**
+     * Finds all matches in the given text according to the specified pattern.
+     *
+     * @param text    the text to search for matches
+     * @param pattern the pattern to search for
+     * @return an unmodifiable list of found matches; returns an empty List if text or pattern is null
+     */
+    public List<String> findMatches(CharSequence text, Pattern pattern) {
+        if (text == null || pattern == null){
+            return Collections.emptyList();
+        }
+
+        List<String> matches = new ArrayList<>();
+        Matcher matcher = pattern.matcher(text);
+
+        while (matcher.find()) {
+            matches.add(matcher.group());
+        }
+
+        return Collections.unmodifiableList(matches);
+    }
+
+    /**
+     * Finds all matches in the given text according to the specified pattern
+     * and returns them as an array of strings.
+     *
+     * @param text  the text to search for matches
+     * @param regex the regular expression pattern
+     * @return an array of found matches; returns an empty array if text or regex is null
+     */
+    public String[] findMatchesAsArray(CharSequence text, String regex) {
```

Comment:
Looking at the remaining methods of this class, each method with signature `(CharSequence, Pattern)` has a corresponding method with signature `(String, String)`. Therefore I would define a method:

```java
public static List<String> findAll(String text, String regex)
```

I don't think it is useful to have both a method returning `List<String>` and `String[]`. The return type should be coherent.